flag_enum.go

```diff
@@ -0,0 +1,144 @@
+package cli
+
+import (
+	"flag"
+	"fmt"
+	"os"
+)
+
+// EnumByName Returns the Enum by name from cli.Context.
+func EnumByName(c *Context, name string) Enum {
+	v := c.Value(name)
+	h := v.(EnumHolder)
```

Comment:
What happens if h is nil ? i.e user requests an enum with invalid name ?

Reply:
I haven't payed attention to it.
I would suggest returning nil, if the name is invalid.
Doing so it would correspond to the existing api of other flags.